server/sonar-web/src/main/js/apps/custom-metrics/components/__tests__/Header-test.tsx

```diff
@@ -17,15 +17,22 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import Marionette from 'backbone.marionette';
-import Template from './templates/metrics-layout.hbs';
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import Header from '../Header';
 
-export default Marionette.LayoutView.extend({
-  template: Template,
+it('should create new metric', () => {
+  const onCreate = jest.fn(() => Promise.resolve());
+  const wrapper = shallow(
+    <Header
+      domains={['Coverage', 'Issues']}
+      loading={false}
+      onCreate={onCreate}
+      types={['INT', 'STRING']}
+    />
+  );
+  expect(wrapper).toMatchSnapshot();
 
-  regions: {
-    headerRegion: '#metrics-header',
-    listRegion: '#metrics-list',
-    listFooterRegion: '#metrics-list-footer'
-  }
+  wrapper.find('CreateButton').prop<Function>('onCreate')({ key: 'foo', name: 'Foo', type: 'INT' });
+  expect(onCreate).toBeCalledWith({ key: 'foo', name: 'Foo', type: 'INT' });
```

Comment:
I'm a bit skeptical about the usefulness of such test, I feel like it's basically calling a function and checking if that function was called...

Reply:
that's because you know the current implementation, but it might change tomorrow...